### DISCUSSION TITLE
Is gfx already dead (or at least "inactive") ?

### DISCUSSION THREAD

**Discussion by DraftedDev** [gfx-rs/gfx#3793] [Q&A] [answered]
# Introduction to my thoughts...
I'm not really into Graphics Programming that much, but I want to get learn about it a bit more for some Project I planned.
I stumbled upon some Graphics Rendering Frameworks (to bundle Vulkan, Metal, Direct, etc into one API), however everything on crates.io is kinda related to gfx (basically just higher level APIs) and since [this](https://github.com/gfx-rs/gfx/discussions/3768) post, I wonder if the "back-bone" of most graphics related crates is already kind of "dead" or at least inactive.
I know the discussion mentioned above covers some future plans, however I don't really know the state of this Project right now and I wonder what advantages I may have by using gfx instead of wgpu, because it seems like everything is switching to wgpu (which is also developed by gfx-rs devs).

# Questions considering the future of gfx...
Is wgpu the "successor" to gfx-rs (the README.md says no, but I doubt it) ?
Does gfx have any "active" & "alive" future or is wgpu just the main crate, when looking at graphics rendering frameworks ?
What does wgpu different, than gfx and why is it overtaking gfx (since it's basically built on top of it) ?

I hope you can cover all my questions.
I am sorry if I said something stupid or stated false facts, feel free to correct me 😄

**Comment by kvark** [comment, marked as answer]
> Is wgpu the "successor" to gfx-rs (the README.md says no, but I doubt it) ?

Yes

> Does gfx have any "active" & "alive" future or is wgpu just the main crate, when looking at graphics rendering frameworks ?

No, gfx doesn't have any future as it stands right now. You can guess by looking at the commit history that nothing has been happening.

> What does wgpu different, than gfx and why is it overtaking gfx (since it's basically built on top of it) ?

wgpu is a slightly higher level abstraction, which is also an official WebGPU standard. So it's better suited (and designed) to run on different backends, unlike gfx's choice of Vulkan-ish API. wgpu is not built on top of gfx any more, instead there is a wgpu-hal layer.

**Reply by DraftedDev** [reply]
So will the gfx/-hal/-etc. crates eventually just be deprecated/unmaintained?
Will there be any {(["real"])}™ successor or at least some kind of "new" Project built from scratch, similar to gfx (maybe xfg?) ?
Anyway, it's sad to see a unique Project™ like this die D:

**Reply by kvark** [reply]
They are effectively deprecated, just not marked like this on crates.
The closest successor to gfx-hal is wgpu-hal.
There is also blade-graphics now, but it's much newer and less polished.
